app/src/main/java/com/nextcloud/talk/serverstatus/ServerStatusRepositoryImpl.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Nextcloud Talk - Android Client
+ *
+ * SPDX-FileCopyrightText: 2025 Sowjanya Kota <sowjanya.kch@gmail.com>
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+package com.nextcloud.talk.serverstatus
+
+import com.nextcloud.talk.api.NcApiCoroutines
+import com.nextcloud.talk.data.user.model.User
+import com.nextcloud.talk.utils.ApiUtils
+import com.nextcloud.talk.utils.database.user.CurrentUserProviderNew
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import javax.inject.Inject
+
+class ServerStatusRepositoryImpl @Inject constructor(
```

Comment:
I believe this should be annotated as a `@Singleton` for dagger to know it should only be instantiated once. [See Docs](https://docs.oracle.com/javaee/7/api/javax/inject/Singleton.html). This way, in case multiple repositories need access to server status information, you can share the instance of the implementation class, and importantly, it's state flow